Solution/source/Menu/Routine.cpp

```diff
@@ -157,33 +154,24 @@ void ThreadMenyooConfig()
 			{
 				MenuConfig::ConfigRead();
 			}
-			bConfigHasNotBeenRead = false;
+			g_ConfigHasNotBeenRead = false;
 		}
 
+		if (GetTickCount() > g_MenyooConfigTick + 30000)
+		{
+			if (MenuConfig::bSaveAtIntervals)
+			{
+				MenuConfig::ConfigSave();
+			}
+			g_MenyooConfigTick = GetTickCount();
+		}
 	}
 }
 
-void ThreadRainbowFader()
+void TickRainbowFader()
 {
-	/*srand(GetTickCount());
-	for (;;)
-	{
-	auto& colour = g_fadedRGB;
-	RgbS& nextColour = RgbS::Random();
-	RgbS twentypercent(0.2f * abs(nextColour.R - colour.R), 0.2f * abs(nextColour.G - colour.G), 0.2f * abs(nextColour.B - colour.B));
-
-	for (UINT8 i = 0; i < 10; i++)
-	{
-	std::this_thread::sleep_for(std::chrono::milliseconds(9));
-	colour.R += twentypercent.R * (nextColour.R > colour.R) ? 1 : (nextColour.R == colour.R) ? 0 : -1;
-	colour.G += twentypercent.G * (nextColour.G > colour.G) ? 1 : (nextColour.G == colour.G) ? 0 : -1;
-	colour.B += twentypercent.B * (nextColour.B > colour.B) ? 1 : (nextColour.B == colour.B) ? 0 : -1;
-	}
-	std::this_thread::sleep_for(std::chrono::milliseconds(340));
-	}*/
-	auto& colour = g_fadedRGB;
-	for (;;)
-	{
+	if (GetTickCount() > g_FaderTick + 20) {
```

Comment:
I'm not sure about this one but wouldn't `if( _programTick % 20 )` work? Cause In that case there's no need for the **g_FaderTick** variable 😄